component/core/src/test/java/org/exoplatform/social/core/jpa/storage/LabelStorageTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ *
+ * Copyright (C) 2023 Meeds Association contact@meeds.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.exoplatform.social.core.jpa.storage;
+
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.jpa.test.AbstractCoreTest;
+import org.exoplatform.social.core.model.Label;
+import org.exoplatform.social.core.storage.api.LabelStorage;
+
+/**
+ * Unit Tests for {@link LabelStorage}
+ */
+public class LabelStorageTest extends AbstractCoreTest {
+
+  private final Log             LOG = ExoLogger.getLogger(LabelStorageTest.class);
+
+  private String objectId = "1";
+  private String objectType = "testObject";
+  private LabelStorage labelStorage;
+  
+  
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    labelStorage = getContainer().getComponentInstanceOfType(LabelStorage.class);
+    assertNotNull(labelStorage);
+  }
+
+  public void testSaveLabel() {
+    try {
+      Label label = createLabel("test label fr", "fr");
+      label = labelStorage.saveLabel(label, true);
+      label = labelStorage.findLabelByObjectTypeAndObjectIdAndLang(objectType,objectId,"fr");
+      assertNotNull(label.getId());
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      fail(e.getMessage());
+    }
+  }
+
+  public void testDeleteLabel() {
+    try {
+      Label label = createLabel("test label fr", "fr");
+      label = labelStorage.saveLabel(label, true);
+      label = labelStorage.findLabelByObjectTypeAndObjectIdAndLang(objectType,objectId,"fr");
+      assertNotNull(label.getId());
+      labelStorage.deleteLabel(label.getId());
+      label = labelStorage.findLabelByObjectTypeAndObjectIdAndLang(objectType,objectId,"fr");
+      assertNull(label);
+    } catch (Exception e) {
```

Comment:
using try/catch in Unit Test isn't relevant anymore, please add throws declaration in method signature and never use try/catch. Either use `assertThrows` to test an exception throw or `throws Exception` in test method signature if not meant to be handled using `assertThrows`.

Reply:
Done